autogpt_platform/backend/backend/blocks/llm.py

```diff
@@ -622,3 +623,217 @@ def run(self, input_data: Input, **kwargs) -> BlockOutput:
             yield "response", response
         except Exception as e:
             yield "error", f"Error calling LLM: {str(e)}"
+
+
+class AIListGeneratorBlock(Block):
+    class Input(BlockSchema):
+        focus: str | None = SchemaField(
+            description="The focus of the list to generate.",
+            placeholder="The top 5 most interesting news stories in the data.",
+            default=None,
+            advanced=False,
+        )
+        source_data: str | None = SchemaField(
+            description="The data to generate the list from.",
+            placeholder="News Today: Humans land on Mars: Today humans landed on mars. -- AI wins Nobel Prize: AI wins Nobel Prize for solving world hunger. -- New AI Model: A new AI model has been released.",
+            default=None,
+            advanced=False,
+        )
+        model: LlmModel = SchemaField(
+            title="LLM Model",
+            default=LlmModel.GPT4_TURBO,
+            description="The language model to use for generating the list.",
+            advanced=True,
+        )
+        api_key: BlockSecret = SecretField(value="")
+        max_retries: int = SchemaField(
+            default=3,
+            description="Maximum number of retries for generating a valid list.",
+            ge=1,
+            le=5,
+        )
+
+    class Output(BlockSchema):
+        generated_list: List[Any] = SchemaField(description="The generated list.")
```

Comment:
Why is it not list of string ?

Reply:
Wouldn't this not allow for nested lists and lists of dictionaries etc?